dropwizard-jdbi3/src/main/java/io/dropwizard/jdbi3/NamePrependingTemplateEngine.java

```diff
@@ -0,0 +1,46 @@
+package io.dropwizard.jdbi3;
+
+import org.jdbi.v3.core.extension.ExtensionMethod;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.TemplateEngine;
+
+import java.util.Objects;
+
+/**
+ * Jdbi {@link TemplateEngine} that prepends the SQLObject's type and method name in front of every
+ * statement sent to the database.
+ */
+public class NamePrependingTemplateEngine implements TemplateEngine {
+    private final TemplateEngine originalEngine;
+
+    /**
+     * @param originalEngine    The original {@link TemplateEngine} to use to create the actual statements
+     */
+    public NamePrependingTemplateEngine(TemplateEngine originalEngine) {
+        this.originalEngine = Objects.requireNonNull(originalEngine, "'originalEngine' cannot be null'");
+    }
+
+    @Override
+    public String render(String template, StatementContext ctx) {
+        final ExtensionMethod extensionMethod = ctx.getExtensionMethod();
+        final String originalResult = originalEngine.render(template, ctx);
+
+        if ((extensionMethod != null)
```

Comment:
I think it would be more clear to make a check that the extension method _doesn't_ exist and exit early.
```java
if (extensionMethod == null || extensionMethod.getType() == null || extensionMethod.getMethod() == null) {
            return originalResult;
}
```

Reply:
Turns out `ExtensionMethod` has `requireNonNull` for type and method so I'll remove those unnecessary checks as well.